tasks/rsync.py

```diff
@@ -0,0 +1,218 @@
+                - suites/iozone.sh
+
+    rsync task sequentially based on runtime. default runtime is 0
+    tasks:
+      - ceph:
+      - ceph-fuse:
+      - rsync:
+            runtime: 120
+
+'''
+
+class RSync(Greenlet):
+
+    def __init__(self, ctx, config, logger):
+        super(RSync, self).__init__()
+
+        self.ctx = ctx
+
+        self.stopping = Event()
+        self.config = config
+        self.logger = logger
+
+        self.my_mnt = None
+        self.workUnit = False
+
+        self.runTime = self.config.get('runtime', 0)
+        self.fileSize = self.config.get('filesize', 2)
+        self.waitTime = self.config.get('waittime', 5)
+
+        self.fs = Filesystem(self.ctx)
+
+        self.snapEnable = bool(self.config.get('snapenable', False))
+        self.logger.info('Snapenable param: {}'.format(self.snapEnable))
+
+        if self.config.get('data_dir'):
+            self.workUnit = True
+            self.dataDir = misc.get_testdir(self.ctx)
+            self.sourceDir = self.dataDir + '/workunit.{}/'.format(self.config.get('data_dir'))
+        else:
+            self.snapShot = None
+            self.snapDummy = None
+            self.iteration = 0
+            self.dataDir = 'source'
+            self.sourceDir = self.dataDir + '/subdir'
+
+    def _run(self):
+        try:
+            self.do_rsync()
+        except:
+            # Log exceptions here so we get the full backtrace (it's lost
+            # by the time someone does a .get() on this greenlet)
+            self.logger.exception("Exception in do_rsync:")
+            raise
+
+    def log(self, x):
+        """Write data to logger assigned to this MDThrasher"""
+        self.logger.info(x)
+
+    def stop(self):
+        time.sleep(self.runTime)
+        self.stopping.set()
+
+    def rsync_io(self):
+        try:
+            if self.snapEnable:
+                # Rsync snapshot data to rsync directory
+                self.my_mnt.run_shell(["rsync", "-azvh", "{}".format(self.snapShot), "rsyncdir/dir1/"])
+            else:
+                # Rsync data from data directory to rsync directory
+                self.my_mnt.run_shell(["rsync", "-azvh", "{}".format(self.sourceDir), "rsyncdir/dir1/"])
+
+        except CommandFailedError:
+            self.logger.error('rsync command failed asserting rsync')
+            raise
+
+    def do_rsync(self):
+
+        # Get mount objects
+        mounts = [v for k, v in sorted(self.ctx.mounts.items(), lambda a, b: cmp(a[0], b[0]))]
+
+        # Using first mount point of cephfs for run rsync test.
+        if len(mounts):
+            self.my_mnt = mounts.pop(0)
+        else:
+            assert len(mounts) > 0, self.logger.error('No mount available asserting rsync')
+
+        if self.snapEnable:
+            # Enable snapshots
+            self.fs.mon_manager.raw_cluster_cmd("mds", "set", "allow_new_snaps", "true", "--yes-i-really-mean-it")
+            self.snapDummy = self.dataDir + '/.snap/snap'
```

Comment:
"snapdummy" variable is only used one place, so get rid of it and just calculate the path in-place below